url39209c0557_au_research_ANDS_RIFCS_API_tgz-pJ8-org_ands_rifcs_example_SampleRIFCSBuildCollectionV1Dot6Number1J8/src/org/ands/rifcs/example/SampleRIFCSRead.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2008 The Australian National University (ANU)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ands.rifcs.example;
+
+import java.io.FileNotFoundException;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.ands.rifcs.base.Collection;
+import org.ands.rifcs.base.Name;
+import org.ands.rifcs.base.NamePart;
+import org.ands.rifcs.base.RIFCS;
+import org.ands.rifcs.base.RIFCSException;
+import org.ands.rifcs.base.RIFCSWrapper;
+import org.ands.rifcs.base.RegistryObject;
+
+import org.ands.rifcs.ch.RIFCSReader;
+
+import org.xml.sax.SAXException;
+
+/** Example of the API that reads RIF-CS data from a file. */
+public final class SampleRIFCSRead {
+    /** The RIF-CS object. */
+    private static RIFCS rifcs = null;
+
+    /** This class can not be instantiated. */
+    private SampleRIFCSRead() {
+    }
+
+    /** The main method.
+     *  @param args The command-line arguments. args[0] is the name
+     *          of a file to be read.
+     *  @throws RIFCSException A RIFCSException
+     *  @throws FileNotFoundException A FileNotFoundException
+     *  @throws SAXException A SAXException
+     *  @throws ParserConfigurationException A ParserConfigurationException
+     *  @throws IOException An IOException
+     *  @throws MalformedURLException A MalformedURLException
+     */
+    public static void main(final String[] args) throws RIFCSException,
+    FileNotFoundException, SAXException, ParserConfigurationException,
+    IOException, MalformedURLException {
+        RIFCSReader rr = new RIFCSReader();
+        rr.mapToDOM(new FileInputStream(args[0]));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when RIFCSReader.mapToDOM throws an exception. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).